digdag-standards/src/main/java/io/digdag/standards/operator/HttpOperatorFactory.java

```diff
@@ -0,0 +1,389 @@
+            }
+
+            return builder
+                    .storeParams(result)
+                    .build();
+        }
+
+        private HttpClient client()
+        {
+            SslContextFactory sslContextFactory = null;
+
+            Optional<Boolean> insecure = params.getOptional("insecure", boolean.class);
+            if (insecure.isPresent()) {
+                sslContextFactory = new SslContextFactory(insecure.get());
+            }
+
+            HttpClient httpClient = new HttpClient(sslContextFactory);
+
+            configureProxy(httpClient);
+
+            boolean followRedirects = params.get("follow_redirects", boolean.class, true);
+
+            httpClient.setFollowRedirects(followRedirects);
+            httpClient.setMaxRedirects(maxRedirects);
+
+            httpClient.setUserAgentField(new HttpField(
+                    USER_AGENT, userAgent + ' ' + httpClient.getUserAgentField().getValue()));
+
+            try {
+                httpClient.start();
+            }
+            catch (Exception e) {
+                throw new TaskExecutionException(e, TaskExecutionException.buildExceptionErrorConfig(e));
+            }
+            return httpClient;
+        }
+
+        private void configureProxy(HttpClient httpClient)
+        {
+            List<ProxyConfiguration.Proxy> proxies = httpClient.getProxyConfiguration().getProxies();
+            Config userProxyConfig = params.getNestedOrGetEmpty("proxy");
+            boolean userProxyEnabled = userProxyConfig.get("enabled", boolean.class, false);
+            if (allowUserProxy && userProxyEnabled) {
+                String host = userProxyConfig.get("host", String.class);
+                int port = userProxyConfig.get("port", int.class);
+                boolean tls = userProxyConfig.get("tls", boolean.class, false);
+                proxies.add(new HttpProxy(new Origin.Address(host, port), tls));
+            }
+            else {
+                if (systemProxy.isPresent()) {
+                    proxies.add(systemProxy.get());
+                }
+                else {
+                    configureEnvProxy("http", proxies);
+                    configureEnvProxy("https", proxies);
+                }
+            }
+        }
+
+        private void stop(HttpClient httpClient)
+        {
+            try {
+                httpClient.stop();
+            }
+            catch (Exception e) {
+                logger.warn("Failed to stop http client", e);
+            }
+        }
+    }
+
+    private void configureEnvProxy(String scheme, List<ProxyConfiguration.Proxy> proxies)
+    {
+        Optional<ProxyConfig> proxyConfig = Proxies.proxyConfigFromEnv(scheme, env);
+        if (proxyConfig.isPresent()) {
+            ProxyConfig c = proxyConfig.get();
+            proxies.add(new SchemeProxy(scheme, new Origin.Address(c.getHost(), c.getPort()), c.useSSL()));
+        }
+    }
+
+    private class SchemeProxy
```

Comment:
The intent here is to specify separate proxy configurations for `http` and `https` requests.

The vanilla `HttpProxy` does not allow that, hence the subclassing and overriding of the `matches` method.